src/nvrtc/safe.rs

```diff
@@ -222,47 +223,147 @@ impl CompileOptions {
     }
 }
 
-pub struct RustPtx {
-    kernel_path: PathBuf,
+#[derive(Debug)]
```

Comment:
RustPtx -> PtxCrate and PtxCrateKind (planned to eventually support standalone compilation directly through rustc. Standalone is possible with

```rust
        let output = Command::new("rustc")
            .arg("+nightly")
            .arg("--crate-type=cdylib")
            .arg("--target")
            .arg("nvptx64-nvidia-cuda")
            .arg(kernel_path) // change the name
            .arg("--")
            .arg("--emit")
            .arg("asm")
            .output()
            .map_err(|e| format!("Failed to execute command: {}", e))?;
```
or some variant of that. It produces massive ptx files. I couldn't figure out how to add the flags -C and opt-level=3

Reply:
Other idea: create a temporary Cargo.toml to accompany a standalone kernel lib.rs.

Reply:
Hmm yeah, I can't quite tell which version developers of kernels would prefer. My instinct is to just choose whichever one is simpler so we aren't over complicated things. Since you've already implemented the cargo variant, maybe we just remove the standalone for now? It will be okay to introduce breaking changes here in the future, since this depends on some very not stable features. It'd also be easier to develop & test if we just have 1 version

Thoughts?

Reply:
Sure, that's cool. Also I like what you did with the command arguments for the C/C++ -> PTX emitter. Definitely a good candidate for future breaking changes to give more features once this has time to settle.

Reply:
Yeah let's just remove it entirely 😄